src/python-common/ceph/deployment/service_spec.py

```diff
@@ -556,6 +545,17 @@ def from_json(cls: Type[ServiceSpecT], json_spec: Dict) -> ServiceSpecT:
 
         return _cls._from_json_impl(c)  # type: ignore
 
+    @staticmethod
+    def validate_json(json_spec: dict) -> dict:
```

Comment:
this function not only validate `json_spec`, it also normalizes it. probably could rename it to `normalize_json()`?